federation/pkg/federation-controller/service/dns/dns.go

```diff
@@ -513,8 +541,11 @@ func (s *ServiceDNSController) ensureDNSRecords(clusterName string, service *v1.
 		"", // nowhere to go up from global level
 	}
 
+	zoneEndpoints, regionEndpoints, globalEndpoints, err := s.getHealthyEndpoints(clusterName, service)
```

Comment:
Just moved this call down near to where we are using these zoneEndpoints, regionEndpoints, globalEndpoints vars.